src/pantry.js

```diff
@@ -0,0 +1,54 @@
+class Pantry {
+  constructor(user, ingredientsData) {
+    this.ingredients = user.pantry;
+    this.ingredientsData = ingredientsData;
+  }
+
+checkIngredients(user, recipe) {
```

Comment:
This is one heck of a method, and the logic tracks really well. It looks readable, even if it is a little long. I'm sure there are places for refactor, but let's call it a win that it does what we need it to do! Stellar work!